src/Microsoft.IdentityModel.Validators/AadIssuerValidator/AadIssuerValidator.cs

```diff
@@ -306,11 +350,58 @@ private static bool IsValidIssuer(string validIssuerTemplate, string tenantId, s
             }
         }
 
-        private BaseConfigurationManager GetEffectiveConfigurationManager(SecurityToken securityToken)
+        private void SetEffectiveLKGIssuer(string aadIssuer, bool isV2Issuer, TimeSpan lastKnownGoodLifetime)
         {
-            var isV2 = securityToken.Issuer.EndsWith(V2EndpointSuffixWithTrailingSlash, StringComparison.OrdinalIgnoreCase) ||
+            var issuerLKG = new IssuerLastKnownGood
+            {
+                Issuer = aadIssuer,
+                LastKnownGoodLifetime = lastKnownGoodLifetime
+            };
+
+            if (isV2Issuer)
+                _issuerLKGV2 = issuerLKG;
+            else
+                _issuerLKGV1 = issuerLKG;
+        }
+
+        private string GetEffectiveLKGIssuer(bool isV2Issuer)
+        {
+            var effectiveLKGIssuer = isV2Issuer ? _issuerLKGV2 : _issuerLKGV1;
+            if (effectiveLKGIssuer != null && effectiveLKGIssuer.IsValid)
+            {
+                return effectiveLKGIssuer.Issuer;
+            }
+
+            return null;
+        }
+
+        private static bool IsV2Issuer(SecurityToken securityToken)
+        {
+            return securityToken.Issuer.EndsWith(V2EndpointSuffixWithTrailingSlash, StringComparison.OrdinalIgnoreCase) ||
                 securityToken.Issuer.EndsWith(V2EndpointSuffix, StringComparison.OrdinalIgnoreCase);
-            return isV2 ? ConfigurationManagerV2 : ConfigurationManagerV1;
+        }
+
+        private BaseConfigurationManager GetEffectiveConfigurationManager(bool isV2Issuer)
+        {        
+            if (_configurationManagerProvider != null)
+            {
+                var aadAuthority = isV2Issuer ? AadAuthorityV2 : AadAuthorityV1;
+                var configurationManager = _configurationManagerProvider(aadAuthority);
+
+                if (configurationManager != null)
```

Comment:
This is interesting to me, what is the expected contract here? Should this ever return null?